assets/ts/api/request.ts

```diff
@@ -0,0 +1,39 @@
+interface requestParameter {
+    endpoint: string,
+    query?: object,
+    body?: object,
+    method?: "GET" | "POST" | "PUT",
+    headers?: object
+}
+
+export function request<dataType>({ endpoint, query, body, method, headers }: requestParameter): Promise<dataType> {
+    const auth = localStorage.getItem("auth-token");
+    if (!auth) {
+        throw new Error("not logged in");
+    }
+
+    const url = new URL(`https://api.spotify.com/v1${endpoint}`);
+    if (query) {
+        for (const [key, value] of Object.entries(query)) {
+            url.searchParams.append(key, value);
+        }
```

Comment:
```suggestion
        Object.entries(query).forEach(([key, value]) => url.searchParams.append(key, value));        
```

Reply:
an sich mag ich forEach nicht, aber ich machs trotzdem

Reply:
Ein kompetenter Kollege von der Arbeit hat mir das mal in einem code review angestrichen, dass .forEach() in javaScript schöner sei. Richtig hinterfragt habe ich es nicht. War nur eine Reflexanmerkung. Danke Trotzdem